dbus/methods_test.go

```diff
@@ -739,9 +754,527 @@ func TestStartStopTransientUnit(t *testing.T) {
 	}
 }
 
+// Ensure that oneshot transient unit starting and stopping works.
+func TestStartStopTransientUnitOneshot(t *testing.T) {
```

Comment:
@jonboulle Yeah, I also think it's too many duplications. Though I just couldn't come up with a clean abstraction between the test cases that differ slightly with each other. Let me think about how to simplify it.